### PR TITLE
Generate changelog for 3.0.0

### PR DESCRIPTION
**Pull request by sergos**
Also, remove unreleased/ entries.

NO_DOC=changelog
NO_TEST=changelog
NO_CHANGELOG=changelog

### CODE REVIEW

File: changelogs/3.0.0.md [hunk 1]
```diff
@@ -0,0 +1,478 @@
...
+* A new option `box.cfg.replicaset_name` allows assigning a human-readable name
+  to the replicaset. It works the same as `box.cfg.replicaset_uuid`. Its value
+  must be the same across all instances of one replicaset. The replicaset name
+  is displayed in `box.info.replicaset.name` (gh-5029).
+
+* A new option `box.cfg.instance_name` allows assigning a human-readable name to
+  the instance. It works the same as `box.cfg.instance_uuid`. Its value must be
+  unique in the replicaset. The instance name is displayed in `box.info.name`.
+  Names of other replicas in the same replicaset are visible in
+  `box.info.replication[id].name` (gh-5029).
```
Comment (andreyaksenov) [inline changelogs/3.0.0.md:116]:
```suggestion
* A new option `box.cfg.replicaset_name` allows assigning a human-readable name
  to the replica set. It works the same as `box.cfg.replicaset_uuid`. Its value
  must be the same across all instances of one replica set. The replica set name
  is displayed in `box.info.replicaset.name` (gh-5029).

* A new option `box.cfg.instance_name` allows assigning a human-readable name to
  the instance. It works the same as `box.cfg.instance_uuid`. Its value must be
  unique in the replicaset. The instance name is displayed in `box.info.name`.
  Names of other replicas in the same replica set are visible in
  `box.info.replication[id].name` (gh-5029).
```

File: changelogs/3.0.0.md [hunk 2]
```diff
@@ -0,0 +1,478 @@
...
+
+* Introduced the `xlog.meta()` method for reading a meta block from an xlog
+  file.
+
+### SQL
+
+* Introduced the `SHOW CREATE TABLE` statement (gh-8098).
+* **[Breaking change]** The `DROP CONSTRAINT` statement has been
+  improved (gh-9112).
+* **[Breaking change]** Names in SQL are now case sensitive. To support backward
```
Comment (andreyaksenov) [inline changelogs/3.0.0.md:189]:
```suggestion
* **[Breaking change]** Names in SQL are now case-sensitive. To support backward
```

File: changelogs/3.0.0.md [hunk 3]
```diff
@@ -0,0 +1,478 @@
...
+    * `fiber_channel_close_mode`
+    * `fiber_slice_default`
+    * `box_cfg_replication_sync_timeout`
+    * `c_func_iproto_multireturn`
+
+  More information on the new behavior can be found on the [Module compat](https://www.tarantool.io/en/doc/latest/reference/reference_lua/compat/) page.
+
+### Console
+
+* Now multiline commands can be used by setting continuation marker
```
Comment (andreyaksenov) [inline changelogs/3.0.0.md:288]:
```suggestion
* Now multiline commands can be used by setting the continuation marker
```

File: changelogs/3.0.0.md [hunk 4]
```diff
@@ -0,0 +1,478 @@
...
+## Bugs fixed
+
+### Core
+
+* **[Breaking change]** The key `box.space._schema['cluster']` is renamed to
+  `'replicaset_uuid'`. That is not expected to be breaking because `_schema` is
+  an internal system space, but the key was visible in public and documented
+  (gh-5029).
+* **[Breaking change]** The table `box.info.cluster` is renamed to
+  `box.info.replicaset`. The behaviour can be reverted using the `compat` option
```
Comment (andreyaksenov) [inline changelogs/3.0.0.md:334]:
```suggestion
  `box.info.replicaset`. The behavior can be reverted using the `compat` option
```

File: changelogs/3.0.0.md [hunk 5]
```diff
@@ -0,0 +1,478 @@
...
+
+* Fixed a bug that allowed to change replicaset UUID via `_schema` space.
+* Fixed a bug when new instances could try to register via an anon instance
+  which previously failed to apply `box.cfg{replication_anon = false}`.
+* Fixed a bug when `box.ctl.demote()` with `box.cfg{election_mode = 'off'}`
+  and an owned synchro queue could simply not do anything (gh-6860).
+* Fixed anonymous replicas not receiving the synchronous transaction queue state
+  during join (gh-9401).
+* Now transactions are not rolled back if the transaction fiber is
+  cancelled when waiting for quorum from replicas (gh-9480).
```
Comment (andreyaksenov) [inline changelogs/3.0.0.md:384]:
```suggestion
  cancelled when waiting for a quorum from replicas (gh-9480).
```

File: changelogs/3.0.0.md [hunk 6]
```diff
@@ -0,0 +1,478 @@
...
+* Fixed a very rare bug when a function deletion could cause an inconsistent
+  state of the database (gh-9426).
+* Now `box.cfg()` correctly interprets the `TT_LISTEN` and `TT_REPLICATION`
+  environment variables with query parameters (gh-9539).
+
+### Http
+
+* Fixed a crash on garbage collection of httpc objects (gh-9283).
+* Fixed a crash due to a race in GC finalizers (gh-9346).
+* Fixed behaviour of GC-finalizer of HTTP client Lua object when chunked HTTP
```
Comment (andreyaksenov) [inline changelogs/3.0.0.md:473]:
```suggestion
* Fixed behavior of GC-finalizer of HTTP client Lua object when chunked HTTP
```

File: changelogs/3.0.0.md [hunk 7]
```diff
@@ -0,0 +1,478 @@
...
+* Fixed a bug when the assertion in `box_promote_qsync` would fail in the
+  debug build mode. The assertion is that at the moment when `box_promote_qsync`
+  is called, no other promote is being executed. It turned out that this
+  assertion is basically incorrect. In the release build mode, this incorrect
+  assumption could potentially lead to writing 2 PROMOTE entries in the same
+  term (gh-9263).
+
+### Replication
+
+* Fixed a bug that allowed to change replicaset UUID via `_schema` space.
```
Comment (andreyaksenov) [inline changelogs/3.0.0.md:376]:
```suggestion
* Fixed a bug that allowed to change a replica set UUID via the `_schema` space.
```

**Review by andreyaksenov — APPROVED** [review]
LGTM, only minor issues

File: changelogs/3.0.0.md [hunk 8]
```diff
@@ -0,0 +1,478 @@
...
+* Added support for sending tuple formats in IPROTO call and eval request
+  arguments. `box_tuple_extension` backward compatibility option can be used to
+  disable receiving tuple formats in IPROTO call and eval request arguments
+  (gh-8633).
+* Added a new `is_sync` parameter to `box.atomic()`. To make the transaction
+synchronous, set the `is_sync` option to `true`. Setting `is_sync = false` is
+prohibited. If any value other than true/nil is set, for example
+`is_sync = "some string"`, then an error will be thrown (gh-8650).
+
+### Cli
```
Comment (Totktonada) [inline changelogs/3.0.0.md:264]: Nit: I would change it to CLI.

File: changelogs/3.0.0.md [hunk 9]
```diff
@@ -0,0 +1,478 @@
...
+* `SHOW CREATE TABLE` no longer supports the DEFAULT clause (gh-8793).
+* `SQL_EXPR` functions can now be set as a default value (gh-8793).
+* A literal set as the default value can no longer have a `+` or `-` sign unless
+  the literal is numeric (gh-8793).
+* **[Breaking change]** SQL now uses the BOX mechanism for default
+  values (gh-8793).
+
+### App
+
+* Added the new `--force-recovery` CLI option (gh-8876).
```
Comment (Totktonada) [inline changelogs/3.0.0.md:201]: This section contains just one item regarding a CLI option. I would merge it into the CLI section.

File: changelogs/3.0.0.md [hunk 10]
```diff
@@ -0,0 +1,478 @@
...
+
+### Box
+
+* Non-system spaces in a snapshot file are now sorted by their IDs.
+  As before, all non-system spaces are stored after system spaces (gh-7954).
+* **[Breaking change]** Dropped support for `IPROTO_CALL16` from `net.box`
+  (i.e., the `call_16` option and the `call_16` method for `net.box`
+  connections) (gh-8636).
```
Comment (Totktonada) [inline changelogs/3.0.0.md:209]: Maybe place all the breaking changes at top on the corresponding sections?

File: changelogs/3.0.0.md [hunk 11]
```diff
@@ -0,0 +1,478 @@
...
+
+### Test
+
+#### Fuzz
+
+* Added a grammar-based SQL fuzzer (gh-4826).
+
+### Uri
+
+* Introduced the `login` and `password` fields in `uri.parse()` (gh-9435).
```
Comment (Totktonada) [inline changelogs/3.0.0.md:314]: Uri -> URI. However, I would move it to Lua -- here we have just one item.

File: changelogs/3.0.0.md [hunk 12]
```diff
@@ -0,0 +1,478 @@
...
+* Introduces the fully temporary space type. It is the same as data-temporary
+  but also has temporary metadata. Temporary spaces can now be created in
+  read_only mode, they disappear after server restart and don't exist on
+  replicas (gh-8323).
+
+### Test
+
+#### Fuzz
+
+* Added a grammar-based SQL fuzzer (gh-4826).
```
Comment (Totktonada) [inline changelogs/3.0.0.md:310]: How about moving it down, after 'build'?

File: changelogs/3.0.0.md [hunk 13]
```diff
@@ -0,0 +1,478 @@
...
+* Fixed HREFK, ALOAD, HLOAD, forwarding vs. `table.clear()`.
+* Fixed snapshot PC when linking to `BC_JLOOP` that was a `BC_RET*`.
+* Fixed dangling references to CType.
+* Ensured returned string is alive in `ffi.typeinfo()`.
+* Fixed the missing initialization of the internal structure, leading to a
+  crash when recording a trace with an allocation of cdata.
+
+### Lua
+
+#### Fio
```
Comment (Totktonada) [inline changelogs/3.0.0.md:425]: I would drop the sub-section (fio), because it has just one item.

File: changelogs/3.0.0.md [hunk 14]
```diff
@@ -0,0 +1,478 @@
...
+### Core
+
+* **[Breaking change]** The key `box.space._schema['cluster']` is renamed to
+  `'replicaset_uuid'`. That is not expected to be breaking because `_schema` is
+  an internal system space, but the key was visible in public and documented
+  (gh-5029).
+* **[Breaking change]** The table `box.info.cluster` is renamed to
+  `box.info.replicaset`. The behaviour can be reverted using the `compat` option
+  `box_info_cluster_meaning`
+  (https://tarantool.io/compat/box_info_cluster_meaning) (gh-5029).
```
Comment (Totktonada) [inline changelogs/3.0.0.md:336]: The page is not created yet: https://github.com/tarantool/doc/issues/3465.

**Review by Totktonada — APPROVED** [review]

File: changelogs/3.0.0.md (hunk 1 above)
Comment (sergos) [reply changelogs/3.0.0.md]: updated (you missed one 4th line from bottom), force pushed

File: changelogs/3.0.0.md (hunk 2 above)
Comment (sergos) [reply changelogs/3.0.0.md]: ditto

File: changelogs/3.0.0.md (hunk 3 above)
Comment (sergos) [reply changelogs/3.0.0.md]: ditto

File: changelogs/3.0.0.md (hunk 4 above)
Comment (sergos) [reply changelogs/3.0.0.md]: ditto

File: changelogs/3.0.0.md (hunk 5 above)
Comment (sergos) [reply changelogs/3.0.0.md]: ditto

File: changelogs/3.0.0.md (hunk 6 above)
Comment (sergos) [reply changelogs/3.0.0.md]: ditto

File: changelogs/3.0.0.md (hunk 7 above)
Comment (sergos) [reply changelogs/3.0.0.md]: ditto

File: changelogs/3.0.0.md (hunk 8 above)
Comment (sergos) [reply changelogs/3.0.0.md]: ditto

File: changelogs/3.0.0.md (hunk 9 above)
Comment (sergos) [reply changelogs/3.0.0.md]: ditto

File: changelogs/3.0.0.md (hunk 10 above)
Comment (sergos) [reply changelogs/3.0.0.md:206]: done

File: changelogs/3.0.0.md (hunk 11 above)
Comment (sergos) [reply changelogs/3.0.0.md]: done, moved to lua

File: changelogs/3.0.0.md (hunk 12 above)
Comment (sergos) [reply changelogs/3.0.0.md]: done

File: changelogs/3.0.0.md (hunk 13 above)
Comment (sergos) [reply changelogs/3.0.0.md]: done

File: changelogs/3.0.0.md (hunk 14 above)
Comment (sergos) [reply changelogs/3.0.0.md:332]: We just started rolling docs. It will appears soon. @andreyaksenov could you double check the address will be valid?

Comment (andreyaksenov) [reply changelogs/3.0.0.md:332]: The `https://tarantool.io/compat/box_info_cluster_meaning` page doesn't exist in docs. It is better to add a link to the root `Module compat` topic: https://www.tarantool.io/en/doc/latest/reference/reference_lua/compat/

Comment (andreyaksenov) [reply changelogs/3.0.0.md:332]: We can update this as follows:

```
The behaviour can be reverted using the `compat` option
  `box_info_cluster_meaning`
  (https://tarantool.io/compat/box_info_cluster_meaning) (gh-5029).
  
->

The behaviour can be reverted using the [compat](https://www.tarantool.io/en/doc/latest/reference/reference_lua/compat/) option
  `box_info_cluster_meaning` (gh-5029).
```

Comment (sergos) [reply changelogs/3.0.0.md:332]: agreed to remove direct link.

**Pull request merged**